src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-list/rbd-list.component.ts

```diff
@@ -0,0 +1,238 @@
+          images = images.concat(pool.value);
+        });
+        const viewCacheStatusList = [];
+        _.forEach(viewCacheStatusMap, (value, key) => {
+          viewCacheStatusList.push({
+            status: parseInt(key, 10),
+            statusFor: (value.length > 1 ? 'pools ' : 'pool ') +
+            '<strong>' + value.join('</strong>, <strong>') + '</strong>'
+          });
+        });
+        this.viewCacheStatusList = viewCacheStatusList;
+        images.forEach(image => {
+          image.executingTasks = this._getExecutingTasks(executingTasks,
+            image.pool_name, image.name);
+        });
+        this.images = this._merge(images, executingTasks);
+        this.executingTasks = executingTasks;
+      },
+      () => {
+        this.viewCacheStatusList = [{status: ViewCacheStatus.ValueException}];
+      }
+    );
+  }
+
+  _getExecutingTasks(executingTasks: ExecutingTask[], poolName, imageName): ExecutingTask[] {
+    const result: ExecutingTask[] = [];
+    executingTasks.forEach(executingTask => {
+      if (executingTask.name === 'rbd/snap/create' ||
+        executingTask.name === 'rbd/snap/delete' ||
+        executingTask.name === 'rbd/snap/edit') {
+        if (poolName === executingTask.metadata['pool_name'] &&
+          imageName === executingTask.metadata['image_name']) {
+          result.push(executingTask);
+        }
+      }
+    });
+    return result;
+  }
+
+  _merge(rbds: RbdModel[], executingTasks: ExecutingTask[] = []) {
+    const resultRBDs = _.clone(rbds);
+    executingTasks.forEach((executingTask) => {
+      if (executingTask.name === 'rbd/create') {
+        const exists = resultRBDs.some((rbd) => {
+          return rbd.pool_name === executingTask.metadata['pool_name'] &&
+            rbd.name === executingTask.metadata['image_name'];
+        });
+        if (!exists) {
+          const rbdModel = new RbdModel();
+          rbdModel.name = executingTask.metadata['image_name'];
+          rbdModel.pool_name = executingTask.metadata['pool_name'];
+          rbdModel.executing = 'creating';
+          resultRBDs.push(rbdModel);
+        }
+      } else if (executingTask.name === 'rbd/delete') {
+        const rbdToDelete = resultRBDs.find((rbd) => {
+          return rbd.pool_name === executingTask.metadata['pool_name'] &&
+            rbd.name === executingTask.metadata['image_name'];
+        });
+        if (rbdToDelete) {
+          rbdToDelete.executing = 'deleting';
+        }
+      } else if (executingTask.name === 'rbd/edit') {
+        const rbdToDelete = resultRBDs.find((rbd) => {
+          return rbd.pool_name === executingTask.metadata['pool_name'] &&
+            rbd.name === executingTask.metadata['image_name'];
+        });
+        if (rbdToDelete) {
+          rbdToDelete.executing = 'updating';
+        }
+      }
+    });
+    return resultRBDs;
+  }
+
+  updateSelection(selection: CdTableSelection) {
+    this.selection = selection;
+  }
+
+  deleteRbdModal() {
```

Comment:
If this was supposed to pop-up before I deleted an image, it didn't seem to work. Probably a good idea to have a delete confirmation modal when deleting a snapshot as well.

Reply:
Will be done after https://github.com/ceph/ceph/pull/20899 is merged.